server.go

```diff
@@ -4217,11 +4217,9 @@ func (s *server) DisconnectPeer(pubKey *btcec.PublicKey) error {
 	delete(s.persistentPeers, pubStr)
 	delete(s.persistentPeersBackoff, pubStr)
 
-	// Remove the current peer from the server's internal state and signal
-	// that the peer termination watcher does not need to execute for this
-	// peer.
-	s.removePeer(peer)
-	s.ignorePeerTermination[peer] = struct{}{}
+	// Remove the peer by calling Disconnect. Previously this was done with
+	// removePeer, which bypassed the peerTerminationWatcher.
+	peer.Disconnect(fmt.Errorf("server: DisconnectPeer called"))
```

Comment:
I think the peer may still exist in `s.ignorePeerTermination`, from `InboundPeerConnected` or `OutboundPeerConnected`. We may need to delete it from `s.ignorePeerTermination` here or remove the callback in `s.scheduledPeerConnection`. Not sure tho, maybe @ellemouton has a better idea from #5700.

Reply:
It's not possible for `s.ignorePeerTermination` to be populated here since:
- the mutex is held
- `Inbound/OutboundPeerConnected` will call `removePeer` and remove from the peer map and also add to `ignorePeerTermination`
  - so `findPeerByPubStr` won't return anything

The original logic was just so that `removePeer` wasn't called in `peerTerminationWatcher` since it was previously called here. But now we want `removePeer` to be called by `peerTerminationWatcher`.

Reply:
Yeah you are right, it's been protected by the `peersByPub`. What happens if this is a persistent peer then? We now leave it to `peerTerminationWatcher` to handle the disconnect, but it will be connected again?

Reply:
previous behavior was to remove the persistent peer from the map, so this doesn't change that (see above the change here)